flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-iceberg/src/main/java/org/apache/flink/cdc/connectors/iceberg/sink/v2/PostCommitOperator.java

```diff
@@ -0,0 +1,127 @@
+import org.apache.flink.streaming.api.operators.OneInputStreamOperator;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.CatalogUtil;
+import org.apache.iceberg.actions.RewriteDataFilesActionResult;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.flink.actions.Actions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/** An Operator to add checkpointId to IcebergTable Small file compaction. */
+public class PostCommitOperator
+        extends AbstractStreamOperator<CommittableMessage<WriteResultWrapper>>
+        implements OneInputStreamOperator<
+                CommittableMessage<WriteResultWrapper>, CommittableMessage<WriteResultWrapper>> {
+    protected static final Logger LOGGER = LoggerFactory.getLogger(PostCommitOperator.class);
+
+    private final Map<String, String> catalogOptions;
+
+    private Catalog catalog;
+
+    /** store a list of MultiTableCommittable in one checkpoint. */
+    private final List<WriteResultWrapper> multiTableCommittables;
+
+    public PostCommitOperator(Map<String, String> catalogOptions) {
+        multiTableCommittables = new ArrayList<>();
+        this.catalogOptions = catalogOptions;
+    }
+
+    @Override
+    public void open() throws Exception {
+        super.open();
+    }
+
+    @Override
+    public void processElement(StreamRecord<CommittableMessage<WriteResultWrapper>> element) {
+        if (element.getValue() instanceof CommittableWithLineage) {
+            multiTableCommittables.add(
+                    ((CommittableWithLineage<WriteResultWrapper>) element.getValue())
+                            .getCommittable());
+        }
+    }
+
+    @Override
+    public void finish() {
+        prepareSnapshotPreBarrier(Long.MAX_VALUE);
+    }
+
+    @Override
+    public void prepareSnapshotPreBarrier(long checkpointId) {
+        if (catalog == null) {
+            this.catalog =
+                    CatalogUtil.buildIcebergCatalog(
+                            "cdc-iceberg-catalog", catalogOptions, new Configuration());
+        }
+        for (int i = 0; i < multiTableCommittables.size(); i++) {
+            WriteResultWrapper multiTableCommittable = multiTableCommittables.get(i);
+            multiTableCommittables.set(
+                    i,
+                    new WriteResultWrapper(
+                            multiTableCommittable.getWriteResult(),
+                            multiTableCommittable.getTableId()));
+        }
+    }
+
+    @Override
+    public void snapshotState(StateSnapshotContext context) throws Exception {
+        super.snapshotState(context);
+        long checkpointId = context.getCheckpointId();
+        if (!multiTableCommittables.isEmpty()) {
```

Comment:
Seems like we will compact in each checkpoint, can we config the compaction strategy?